tests/unit/sync/transformTaxonomyGroupModel.test.ts

```diff
@@ -0,0 +1,76 @@
+import { describe, expect, it } from "@jest/globals";
+import { TaxonomyContracts } from "@kontent-ai/management-sdk";
+
+import { transformTaxonomyGroupsModel } from "../../../src/modules/sync/modelTransfomers/taxonomyGroups";
+
+const input: ReadonlyArray<TaxonomyContracts.ITaxonomyContract> = [
+  {
+    id: "taxonomyGroup!Id",
+    last_modified: "groupModified1",
+    name: "taxonomyGroup1",
+    codename: "taxonomy_group1",
+    external_id: "extId_taxonomy_group1",
+    terms: [
+      {
+        id: "nestedTermId1",
+        name: "nestedTerm",
+        codename: "nested_term",
+        last_modified: "nestedTermModofied",
+        terms: [],
+      },
+    ],
+  },
+  {
+    id: "taxonomyGroup2Id",
+    last_modified: "groupModified2",
+    name: "taxonomyGroup2",
+    codename: "taxonomy_group2",
+    external_id: "extId_taxonomy_group2",
+    terms: [
+      {
+        id: "nestedTermId2",
+        name: "nestedTerm2",
+        codename: "nested_term2",
+        last_modified: "nestedTermModofied2",
+        external_id: "extId_nested_term2",
+        terms: [],
+      },
+    ],
+  },
+] as const;
```

Comment:
The `as const` doesn't do anything here, because you assigned a type to the `input` constant explicitly. If you want to infer the specific type (i.e. if you want this to be a tuple of two groups) and still be a subtype of the `ReadonlyArray<TaxonomyContracts.ITaxonomyContract>`, you must use the `satisfies` keyword like this `as const satisfies ReadonlyArray<TaxonomyContracts.ITaxonomyContract>` and remove the explicit type of the `input` constant.